### PR TITLE
Read securityconfig from extension

### PR DESCRIPTION
**Pull request by AmaliMatharaarachchi**
### Purpose

1. added extension support for endpoint security config. 
2. added test cases covering scenarios related to endpoint basic auth security.

### Issues

Fixes https://github.com/wso2/product-microgateway/issues/1945

### Automation tests
 - Unit tests added: No
 - Integration tests added: Yes

### Tested environments

Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

File: adapter/internal/oasparser/model/mgw_swagger.go [hunk 1]
```diff
@@ -712,6 +730,35 @@ func (swagger *MgwSwagger) getEndpoints(vendorExtensions map[string]interface{},
 	return nil, nil // the vendor extension for prod or sandbox just isn't present
 }
 
+// SetEndpointSecurity set endpoint security by reading api.yaml and environment variables
+func (swagger *MgwSwagger) SetEndpointSecurity(apiHashValue string, epSecurityFromYaml APIEndpointSecurity) {
+	if epSecurityFromYaml.Production.Enabled {
```
Comment (AmaliMatharaarachchi) [inline adapter/internal/oasparser/model/mgw_swagger.go:735]: if swagger already given, dont read yaml

**Review by pubudu538 — APPROVED** [review]

**Comment by AmaliMatharaarachchi** [comment]
making this pr a draft till integration test are added

**Comment by AmaliMatharaarachchi** [comment]
Added test cases as well. Ready to merge

File: integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/testcases/withapim/BackendSecurityTestCase.java [hunk 2]
```diff
@@ -122,14 +128,30 @@ public void testBasicAuthSecuredBackend() throws CCTestException, MalformedURLEx
         Assert.assertTrue(respHeaders.containsKey("authorization"), "Backend did not receive auth header");
         Assert.assertEquals(respHeaders.get("authorization"), "Basic YWRtaW46YWRtaW4=",
                 "backend basic auth header is incorrect");
+    }
 
-        //test 2 - invoke non secured resource
-        endpoint = Utils.getServiceURLHttps(API_CONTEXT + "/1.0.0/echo2");
-        response = HttpsClientRequest.doGet(endpoint);
+    @Test(description = "oauth2 secured resource backend basic auth")
```
Comment (renuka-fernando) [inline integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/testcases/withapim/BackendSecurityTestCase.java:133]: the same description of the previous test case (line 118)

Comment (AmaliMatharaarachchi) [inline integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/testcases/withapim/BackendSecurityTestCase.java:133]:
```suggestion
    @Test(description = "test env variables is working")
```

Comment (AmaliMatharaarachchi) [reply integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/testcases/withapim/BackendSecurityTestCase.java]: fixed

**Comment by choreo-cicd** [comment]
[failed] dev-deployment : 20211215.32

**Comment by choreo-cicd** [comment]
[succeeded] dev-deployment : 20211215.43

**Pull request merged**